File: support_tools/swigwin-3.0.12/Examples/test-suite/python/director_detect_runme.py
import director_detect


class MyBar(director_detect.Bar):

    def __init__(self, val=2):
        director_detect.Bar.__init__(self)
        self.val = val

    def get_value(self):
        self.val = self.val + 1
        return self.val

    def get_class(self):
        self.val = self.val + 1
        return director_detect.A()

    def just_do_it(self):
        self.val = self.val + 1

    def clone(self):
        return MyBar(self.val)
    pass


b = MyBar()

f = b.baseclass()

v = f.get_value()
a = f.get_class()
f.just_do_it()

c = b.clone()
vc = c.get_value()

if (v != 3) or (b.val != 5) or (vc != 6):
    raise RuntimeError, "Bad virtual detection"

File: support_tools/swigwin-3.0.12/Examples/test-suite/php/director_detect_runme.php
<?php

require "tests.php";
require "director_detect.php";

// No new functions
check::functions(array(foo_cloner,foo_get_value,foo_get_class,foo_just_do_it,bar_baseclass,bar_cloner,bar_get_value,bar_get_class,bar_just_do_it));
// No new classes
check::classes(array(A,Foo,Bar));
// now new vars
check::globals(array());

class MyBar extends Bar {
  function __construct($val = 2) {
    parent::__construct();
    $this->val = $val;
  }

  function get_value() {
    $this->val = $this->val + 1;
    return $this->val;
  }

  function get_class() {
    $this->val = $this->val + 1;
    return new A();
  }

  function just_do_it() {
    $this->val = $this->val + 1;
  }

  /* clone is a reserved keyword */
  function clone_() {
    return new MyBar($this->val);
  }
}

$b = new MyBar();

$f = $b->baseclass();

$v = $f->get_value();
$a = $f->get_class();
$f->just_do_it();

$c = $b->clone_();
$vc = $c->get_value();

check::equal($v, 3, "f: Bad virtual detection");
check::equal($b->val, 5, "b: Bad virtual detection");
check::equal($vc, 6, "c: Bad virtual detection");

check::done();
?>
